### PR TITLE
Use datetime2 comparisons for incremental fetching

### LINKED ISSUE #74: Stabilize incremental fetching tests
On travis the test `testIncrementalFetchingByTimestamp` fails around half the time on https://github.com/keboola/db-extractor-mssql/blob/master/tests/Keboola/DbExtractor/IncrementalFetchingTest.php#L33
Where it fails asserting that importedRows is `0`, it has `6` instead.  
This is most likely because a leftover output file from the previous run hasn't been removed in time.

### PR DESCRIPTION
**Pull request by pivnicek**
Fixes #74

### CODE REVIEW

**Comment by pivnicek** [comment]
@tomasfejfar I caught this as the tests were failing only sometimes, but too often.  It was because of the incorrect comparison.  Can you please give it a quick look?  Thank you.

File: src/Keboola/DbExtractor/Extractor/MSSQL.php [hunk 1]
```diff
@@ -673,7 +673,7 @@ private function getIncrementalQueryAddon(): ?string
                     );
                 } else if ($this->incrementalFetching['type'] === self::TYPE_TIMESTAMP) {
                     $incrementalAddon = sprintf(
-                        " WHERE %s > '%s'",
+                        " WHERE CONVERT(DATETIME2, %s) > CONVERT(DATETIME2, '%s')",
```
Comment (tomasfejfar) [inline src/Keboola/DbExtractor/Extractor/MSSQL.php]: I'm suspicious regarding the third parameter (https://docs.microsoft.com/en-us/sql/t-sql/functions/cast-and-convert-transact-sql?view=sql-server-2017#date-and-time-styles). Especially about the part 

> style
> An integer expression that specifies how the CONVERT function will translate expression. **For a style value of NULL, NULL is returned.** data_type determines the range.

Comment (tomasfejfar) [reply src/Keboola/DbExtractor/Extractor/MSSQL.php]: I can't seem to make it work locally (connect to docker MSSQL), so I can't test it.

Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/MSSQL.php]: Regarding the 3rd parameter, there's an issue for that in the general query formulation (https://github.com/keboola/db-extractor-mssql/issues/63)
But here, as it is doing a comparison, as long as it is the same for both sides it should work, but I should check this for sure, thanks!  I'll make a new test for some strangely formatted datetime column

Comment (tomasfejfar) [reply src/Keboola/DbExtractor/Extractor/MSSQL.php]: I was worried about the NULL part. The parameter is missing, so it's NULL (I think), so the value of the conversion will probably be NULL as well. 

I now managed to run the server. 
`SELECT (CONVERT(datetime2, '2018-01-01 10:10:12') > CONVERT(datetime2, '2018-01-01 10:10:13')) AS result;`
`/* SQL Error (102): Incorrect syntax near '>'. */`

Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/MSSQL.php]: yeah, sql server doesn't like conditionals in the field portion of queries.

Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/MSSQL.php]: imho, the fact that this test passes means that the date comparison is working correctly.  Are there any cases in that test that I'm missing? maybe we can beef it up?

Comment (tomasfejfar) [reply src/Keboola/DbExtractor/Extractor/MSSQL.php]: You're right. "If it works, don't try to fix it".

**Comment by pivnicek** [comment]
Hmm, I say that, but that test isn't passing....

**Comment by pivnicek** [comment]
You're right of course :/.  I couldn't get it to fail locally.  I'll try again.

**Pull request merged**